### PR TITLE
Refactor deployContexts to StringKeyMap configContexts

### PR DESCRIPTION
**Pull request by paul-thompson-helix**
* Rename  `DeployContextSource` to `ConfigContextSource`

* Define and use `StringKeyMap` in ADL definitions instead of `StringMap` in order to name the keys , Including on maps in Config and State.

* Add new `StringKeyMap<StaticConfigTopicName, ConfigContextSource> configContexts = {};`  to replace `Vector<DeployContext> deployContexts` where `DeployContext = Struct { String name; ConfigContextSource; }`

* Drop config.adl `DeployContext`
* Rename DeployConfigTopicName to StaticConfigTopicName
* Rename fetchDeployContext to fetchConfigContext

* Drop backwards compatibility for original `Toolconfig.deployContexts` field

### CODE REVIEW

**Comment by paul-thompson-helix** [comment]
Extracted from work in progress on dynamic config.  

NB: Requires testing.

**Comment by paul-thompson-helix** [comment]
Rebasing ahead of current origin/master

File: adl/config.adl [hunk 1]
```diff
@@ -23,7 +25,9 @@ struct ToolConfig {
     /// The storage location for release zip files
     BlobStoreConfig releases;
 
-    Vector<DeployContext> deployContexts = [];
+    /// Static Config contexts
+    StringKeyMap<StaticConfigTopicName, ConfigContextSource> configContexts = {};
```
Comment (timbod7) [inline adl/config.adl]: A couple of suggested renames: 

    - StaticConfigTopicName => StaticConfigName
    - ConfigContextSource => JsonSource

**Comment by paul-thompson-helix** [comment]
Added a self contained demo of hx-deploy-tool & a basic config templating example.

Useful to test this branch & for doco generally

**Pull request merged**